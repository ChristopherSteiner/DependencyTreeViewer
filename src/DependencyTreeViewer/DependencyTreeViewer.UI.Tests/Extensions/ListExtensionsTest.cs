using DependencyTreeViewer.Application.Extensions;
using DependencyTreeViewer.Application.NugetTree;
using Snapshooter.Xunit;
using Xunit;
using Xunit.Abstractions;

namespace DependencyTreeViewer.UI.Tests.Extensions;

public class ListExtensionsTest
{
    private readonly List<ProjectModel> projects = new List<ProjectModel>();

    public ListExtensionsTest(ITestOutputHelper testOutput)
    {
        projects = new List<ProjectModel>
        {
            new ProjectModel
            {
                Name = "Foo.csproj",
                TargetFrameworks = new List<FrameworkModel>
                {
                    new FrameworkModel
                    {
                        Name = "net8.0",
                        References = new List<PackageModel>
                        {
                            new PackageModel
                            {
                                Name = "Some.Library",
                                Version = "1.0.0",
                                References = new List<PackageModel>
                                {
                                    new PackageModel
                                    {
                                        Name = "Some.Library.Abc",
                                        Version = "1.3.0",
                                        References = new List<PackageModel>
                                        {
                                            new PackageModel { Name = "Some.Library.Abc.Abstractions", Version = "1.4.1" },
                                            new PackageModel { Name = "Some.Library.Cba", Version = "1.5.3" }
                                        }
                                    },
                                    new PackageModel
                                    {
                                        Name = "Some.Library.Def",
                                        Version = "2.3.1",
                                        References = new List<PackageModel>
                                        {
                                            new PackageModel { Name = "Some.Library.Def.Abstractions", Version = "1.1.0" }
                                        }
                                    }
                                }
                            },
                            new PackageModel
                            {
                                Name = "Other.Library",
                                Version = "2.3.1"
                            },
                            new PackageModel
                            {
                                Name = "Default.Class",
                                Version = "3.0.0",
                                References = new List<PackageModel>
                                {
                                    new PackageModel { Name = "Default.Class.Abstractions", Version = "3.0.0" }
                                }
                            }
                        }
                    }
                }
            },
            new ProjectModel
            {
                Name = "Bar.csproj",
                TargetFrameworks = new List<FrameworkModel>
                {
                    new FrameworkModel
                    {
                        Name = "net8.0",
                        References = new List<PackageModel>
                        {
                            new PackageModel
                            {
                                Name = "Other.Library",
                                Version = "4.3.1"
                            },
                            new PackageModel
                            {
                                Name = "Unit.Library",
                                Version = "18.8.6",
                                References = new List<PackageModel>
                                {
                                    new PackageModel { Name = "Test.Library", Version = "12.1.2" },
                                    new PackageModel
                                    {
                                        Name = "Default.Class",
                                        Version = "3.0.0",
                                        References = new List<PackageModel>
                                        {
                                            new PackageModel { Name = "Default.Class.Abstractions", Version = "3.0.0" }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        };
    }

    [Fact]
    public void Verify_Search_HitOnLastNoteInTree()
    {
        List<ProjectModel> result = projects.Search("cba").ToList();

        Snapshot.Match(result);
    }

    [Fact]
    public void Verify_Search_HitSomewhereInTheMiddleOfTheTree()
    {
        List<ProjectModel> result = projects.Search("Unit").ToList();

        Snapshot.Match(result);
    }

    [Fact]
    public void Verify_Search_MultipleHits()
    {
        List<ProjectModel> result = projects.Search("Library").ToList();

        Snapshot.Match(result);
    }

}
