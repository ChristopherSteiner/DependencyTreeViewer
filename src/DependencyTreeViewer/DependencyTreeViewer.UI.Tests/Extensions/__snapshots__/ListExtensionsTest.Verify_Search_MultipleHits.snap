[
  {
    "Name": "Foo.csproj",
    "FullPath": "",
    "TargetFrameworks": [
      {
        "Name": "net8.0",
        "References": [
          {
            "Name": "Some.Library",
            "Version": "1.0.0",
            "References": [
              {
                "Name": "Some.Library.Abc",
                "Version": "1.3.0",
                "References": [
                  {
                    "Name": "Some.Library.Abc.Abstractions",
                    "Version": "1.4.1",
                    "References": []
                  },
                  {
                    "Name": "Some.Library.Cba",
                    "Version": "1.5.3",
                    "References": []
                  }
                ]
              },
              {
                "Name": "Some.Library.Def",
                "Version": "2.3.1",
                "References": [
                  {
                    "Name": "Some.Library.Def.Abstractions",
                    "Version": "1.1.0",
                    "References": []
                  }
                ]
              }
            ]
          },
          {
            "Name": "Other.Library",
            "Version": "2.3.1",
            "References": []
          }
        ]
      }
    ]
  },
  {
    "Name": "Bar.csproj",
    "FullPath": "",
    "TargetFrameworks": [
      {
        "Name": "net8.0",
        "References": [
          {
            "Name": "Other.Library",
            "Version": "4.3.1",
            "References": []
          },
          {
            "Name": "Unit.Library",
            "Version": "18.8.6",
            "References": [
              {
                "Name": "Test.Library",
                "Version": "12.1.2",
                "References": []
              }
            ]
          }
        ]
      }
    ]
  }
]
